sdk/lib/crt/process/_system.c

```diff
@@ -14,110 +14,85 @@
 #include <process.h>
 
 
-/*
- * @implemented
- */
-int system(const char *command)
-{
-  char *szCmdLine = NULL;
-  char *szComSpec = NULL;
-
-  PROCESS_INFORMATION ProcessInformation;
-  STARTUPINFOA StartupInfo;
-  char *s;
-  BOOL result;
-
-  int nStatus;
-
-  szComSpec = getenv("COMSPEC");
+wchar_t *msvcrt_wstrdupa(const char *); //file.c
+intptr_t do_spawnW(int mode, const wchar_t* cmdname, const wchar_t* args, const wchar_t* envp); //process.c
 
-// system should return 0 if command is null and the shell is found
+/* INTERNAL: retrieve COMSPEC environment variable */
+static wchar_t *get_comspec(void)
+{
+  static const wchar_t cmd[] = {'c','m','d',0};
+  static const wchar_t comspec[] = {'C','O','M','S','P','E','C',0};
+  wchar_t *ret;
+  unsigned int len;
 
-  if (command == NULL) {
-    if (szComSpec == NULL)
-      return 0;
-    else
-      return 1;
+  if (!(len = GetEnvironmentVariableW(comspec, NULL, 0))) len = sizeof(cmd)/sizeof(wchar_t);
+  if ((ret = HeapAlloc(GetProcessHeap(), 0, len * sizeof(wchar_t))))
+  {
+    if (!GetEnvironmentVariableW(comspec, ret, len)) strcpyW(ret, cmd);
```

Comment:
cmd without extension?

Reply:
ComSpec=C:\WINDOWS\system32\cmd.exe

Reply:
I meant, the fall back string "cmd" (that you copy using strcpyW) does not contain any extension.

Reply:
addressed